src/main/java/org/opensearch/neuralsearch/common/MinClusterVersionUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.common;
+
+import com.google.common.collect.ImmutableMap;
+import org.opensearch.Version;
+import org.opensearch.knn.index.IndexUtil;
+import org.opensearch.neuralsearch.util.NeuralSearchClusterUtil;
+
+import java.util.Map;
+
+import static org.opensearch.knn.index.query.KNNQueryBuilder.MAX_DISTANCE_FIELD;
+import static org.opensearch.knn.index.query.KNNQueryBuilder.MIN_SCORE_FIELD;
+import static org.opensearch.neuralsearch.query.NeuralQueryBuilder.MODEL_ID_FIELD;
+
+public class MinClusterVersionUtil {
+
+    private static final Version MINIMAL_SUPPORTED_VERSION_DEFAULT_MODEL_ID = Version.V_2_11_0;
+    private static final Version MINIMAL_SUPPORTED_VERSION_RADIAL_SEARCH = Version.V_2_14_0;
+
+    // Note this minimal version will act as a override
+    private static final Map<String, Version> MINIMAL_VERSION_NEURAL = ImmutableMap.<String, Version>builder()
```

Comment:
nit: I think "NEURAL" prefix is redundant, in plugin repo everything is related to this plugin